modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/friendly/url/resolver/FileEntryFriendlyURLResolverImpl.java

```diff
@@ -35,7 +38,7 @@
 
 	@Override
 	public Optional<FileEntry> resolveFriendlyURL(
-			long groupId, String friendlyURL)
+			long groupId, boolean checkPermissions, String friendlyURL)
```

Comment:
@AliciaGarciaGarcia I really dislike this. From my point of view, the correct approach would be:
1. Let the friendly url resolver work with local services. Its purpose is to resolve a friendly URL to a file entry, nothing else, nothing more. No permission checks, no expiration date checks. This will make it simpler and reusable in other contexts.
2. Make all the checks in the caller. In this case, the web server servlet should ensure that only non expired file entries for which the user has view permissions are visible.

Change the _checkExpiredFileEntry method so that it checks both the expiration date and the VIEW permission (you should rename it as well).